src/lib/crypto/kimchi_backend/gadgets/bitwise.ml

```diff
@@ -708,6 +708,16 @@ let%test_unit "bitwise xor gadget" =
       let cs, _proof_keypair, _proof =
         Runner.generate_and_verify_proof ?cs (fun () ->
             let open Runner.Impl in
+            (* Create half a generic to force a possible generic in the middle *)
```

Comment:
This is fine for a sanity-check, but doesn't this just shift us from checking 1 alignment (an even number of half-generics) to the other (an odd number). This doesn't feel very disciplined.

Reply:
Oh well that's because Xor alone only has 1 generic gate, and thus would only insert a row if it already had 1 pending. But I get your point, in case the generic gate changed again, and had an even number of generics inside, let's modify the tests so it does it for both even and odd preceding generics.